polkadot-protocol/basics/blocks-transactions-fees/transactions.md

```diff
@@ -1,4 +1,253 @@
+
+Signed extensions can enforce checks like:
+
+- [**`CheckSpecVersion`**](https://paritytech.github.io/polkadot-sdk/master/src/frame_system/extensions/check_spec_version.rs.html){target=\_blank} - ensures the transaction is compatible with the runtime's current version
+- [**`CheckWeight`**](https://paritytech.github.io/polkadot-sdk/master/frame_system/struct.CheckWeight.html){target=\_blank} - calculates the weight (or computational cost) of the transaction, ensuring the block doesn't exceed the maximum allowed weight
+
+These extensions are critical in the transaction lifecycle, ensuring that only valid and prioritized transactions are processed.
+
+## Transaction Construction
+
+Building transactions in the Polkadot SDK involves constructing a payload that can be verified, signed, and submitted for inclusion in a block. Each runtime in the Polkadot SDK has its own rules for validating and executing transactions, but there are common patterns for constructing a signed transaction.
+
+### Construct a Signed Transaction
+
+A signed transaction in the Polkadot SDK includes various pieces of data to ensure security, prevent replay attacks, and prioritize processing. Here's an overview of how to construct one:
+
+1. **Construct the unsigned payload** - gather the necessary information for the call, including:
+    - **Pallet index** - identifies the pallet where the runtime function resides
+    - **Function index** - specifies the particular function to call in the pallet
+    - **Parameters** - any additional arguments required by the function call
+2. **Create a signing payload** - once the unsigned payload is ready, additional data must be included:
+    - **Transaction nonce** - unique identifier to prevent replay attacks
+    - **Era information** - defines how long the transaction is valid before it's dropped from the pool
+    - **Block hash** - ensures the transaction doesn't execute on the wrong chain or fork
+3. **Sign the payload** - using the sender's private key, sign the payload to ensure that the transaction can only be executed by the account holder
+4. **Serialize the signed payload** - once signed, the transaction must be serialized into a binary format, ensuring the data is compact and easy to transmit over the network
+5. **Submit the serialized transaction** - finally, submit the serialized transaction to the network, where it will enter the transaction pool and wait for processing by an authoring node
+
+The following is an example of how a signed transaction might look:
+
+``` rust
+--8<-- 'code/polkadot-protocol/basics/blocks-transactions-fees/transactions/signed-tx-example.rs'
+```
+
+### Extrensic Encoding
+
+Before a transaction is sent to the network, it is serialized and encoded using a structured encoding process that ensures consistency and prevents tampering:
+
+- `[1]` - compact encoded length in bytes of the entire extrinsic
```

Comment:
This feels like maybe it should be transaction instead of extrinsic, but not sure